xhttp/client.go

```diff
@@ -103,18 +145,19 @@ func (r *retrierClient) do(ctx context.Context, req *http.Request, requestBody [
 		// For connections reset...same problem:
 		// - https://github.com/golang/go/blob/d0dc93c8e1a5be4e0a44b7f8ecb0cb1417de50ce/src/net/http/transport_test.go#L2207
 		// We need to go for some suffix matches.
-		if strings.Contains(err.Error(), "http2: server sent GOAWAY and closed the connection") ||
+		if errors.Is(err, context.DeadlineExceeded) ||
+			strings.Contains(err.Error(), "http2: server sent GOAWAY and closed the connection") ||
 			strings.HasSuffix(err.Error(), ": connection reset by peer") {
-
+			r.logger.Debug("xhttp.Client: retrying request with error", "error", err, "sleep_period", sleepPeriod)
```

Comment:
We have the log in the ctx, right? Wouldn't be better to use it instead. Something that I would be missing here is the trace_id for example, getting from the ctx we have all the information we need otherwise the log will be lose

Reply:
I'm just not sure if we need to avoid the log dependency, other packages inside of the lib already depend on it, and the lib (even though is public is focused on our use-cases, don't remember if we decide that or not), so I'd be ok to have the dependency. Just feel that will be really easy to miss the log as we would not have the trace-id to match. I was also ok to not having logging TBH.

The hook seems a interesting approach, maybe even the retry codes could be one hook, or kind of middleware that chained decide if retry or not, giving a lot of control for the caller. Or maybe too much things 😅

Reply:
I would tend to just have it depend on our slog package to keep it simple.